app/controllers/organizations_controller.rb

```diff
@@ -80,12 +80,12 @@ def create
   private
 
   def find_organization
-    if current_user && user_is_admin?
-      @organization = Organization.find params[:id]
+    @organization = Organization.find params[:id]
+    if current_user && user_is_admin? || (current_user && current_user.organization_id == @organization.id)
```

Comment:
Hmm works for us when we try as Admin, Owner, User and guest. We'll write tests and confirm.